contracts/modules/v2/StakingPool.sol

```diff
@@ -507,6 +507,131 @@ contract StakingPool is IStakingPool, ERC721 {
+    require(tokenId != 0, "StakingPool: Invalid token id");
+    require(
+      _isApprovedOrOwner(msg.sender, tokenId),
+      "StakingPool: Not authorized to extend deposits on this token"
+    );
+    require(
+      initialTrancheId < newTrancheId,
+      "StakingPool: The chosen tranche cannot end before the initial one"
+    );
+    require(newTrancheId <= maxTranche, "StakingPool: The chosen tranche is not available yet");
+    require(
+      newTrancheId >= _firstActiveTrancheId,
+      "StakingPool: The chosen tranche has already reached the maturity date"
+    );
+
+    // If the intial tranche is expired, withdraw everything and make a new deposit equal to the
+    // withdrawn stake amount plus a top up amount if applicable. This will require the user to
+    // grant sufficient allowance beforehand.
+    if (initialTrancheId < _firstActiveTrancheId) {
+      uint[] memory trancheIds = new uint[](1);
+      trancheIds[0] = initialTrancheId;
+
+      WithdrawParams[] memory withdrawParams = new WithdrawParams[](1);
+      withdrawParams[0] = WithdrawParams(
+        tokenId,
+        true, // Withdraw deposit
+        true, // Withdraw rewards
+        trancheIds
+      );
+
+      (uint withdrawnStake, /* uint rewardsToWithdraw */) = withdraw(withdrawParams);
+
+      depositTo(
+        withdrawnStake + topUpAmount,
+        newTrancheId,
+        tokenId,
+        msg.sender
+      );
+
+      return; // Done! Skip the rest of the function.
+    }
+
+    // If the initial tranche is still active, move all the shares and pending rewards to the
+    // newly chosen tranche and its coresponding deopsit.
+
+    // First make sure tranches are up to date in terms of accumulated NXM rewards.
+    updateTranches();
+
+    Tranche memory initialTranche = tranches[initialTrancheId];
+    Tranche memory newTranche = tranches[newTrancheId];
+    Deposit memory initialDeposit = deposits[tokenId][initialTrancheId];
+    Deposit memory newDeposit = deposits[tokenId][newTrancheId];
+    uint _accNxmPerRewardsShare = accNxmPerRewardsShare;
+
+
+    // [todo] Reward shares should account for inflating the shares, using
+    // calculateRewardSharesAmount and based on the difference between the
+    // first active tranche and the new tranche.
+
+    // The user's shares are moved from the initial tranche to the new one.
+    initialTranche.stakeShares -= initialDeposit.stakeShares;
+    initialTranche.rewardsShares -= initialDeposit.rewardsShares;
+    newTranche.stakeShares += initialDeposit.stakeShares;
+    newTranche.rewardsShares += initialDeposit.rewardsShares;
+
+    // Store the updated tranches.
+    tranches[initialTrancheId] = initialTranche;
+    tranches[newTrancheId] = newTranche;
+
+    // Calculate the rewards that will be carried from the initial deposit to the next one.
+    uint rewardsToCarry;
+    {
+      uint newEarningsPerShare = _accNxmPerRewardsShare - initialDeposit.lastAccNxmPerRewardShare;
```

Comment:
let's put the variable

`_accNxmPerRewardsShare` closer to where it's used, before this bracket